### PR TITLE
Switch to Go. Unit tests

### PR DESCRIPTION
**Pull request by vlpolak**
Switch to Go. Unit tests

### CODE REVIEW

File: server/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+func TestLoginHandlerFunct(t *testing.T) {
+	defer gock.Off()
+	gock.New("localhost:8080").
+		Post("/login").
+		Reply(200).
+		JSON(map[string]string{"user": "Jinzhu", "password": "122234234"})
+
+	body := bytes.NewBuffer([]byte(`{"user": "Jinzhu","password": "122234234"}`))
+	res, err := http.Post("http://127.0.0.1:8080/login/", "application/json", body)
```
Comment (ssh2go) [inline server/handler_test.go:20]: What's about the negative case? For example, wrong user/password or incorrect request

**Review by ssh2go — COMMENTED** [review]
There are no unit-tests for most of the files/functions, could you please implement some? At least:
"github.com/vlpolak/swtgo/cache"
"github.com/vlpolak/swtgo/pkg/hasher"
"github.com/vlpolak/swtgo/server"

And nice to have:
"github.com/vlpolak/swtgo/application"
"github.com/vlpolak/swtgo/infrastructure/persistence"

File: server/handler.go [hunk 2]
```diff
@@ -128,6 +128,11 @@ func (s *Server) LoginHandlerFunc(w http.ResponseWriter, r *http.Request) {
 	user := r.FormValue("user")
 	password := r.FormValue("password")
 
+	err = validateRequest(r)
+	if err != nil {
+		return
```
Comment (ssh2go) [inline server/handler.go:133]: It's better to return error in case when exec result can be different for `LoginHandlerFunc`, or at least log that error

**Review by ssh2go — APPROVED** [review]
Please fix one minor thing and feel free to merge the PR

File: server/handler.go (hunk 2 above)
Comment (vlpolak) [reply server/handler.go:133]: ok

**Pull request merged**